### DISCUSSION TITLE
posts/2021-11-24_using-giscus/

### DISCUSSION THREAD

**Discussion by giscus** [joelnitta/joelnitta-home#6] [Comments]
# posts/2021-11-24_using-giscus/

How to use the giscus commenting system on a Distill blog

https://www.joelnitta.com/posts/2021-11-24_using-giscus/

**Comment by shannonpileggi** [comment]
Hi Joel!  Thanks for this great post! I was just considering switching to utterances, and I hadn't yet heard of giscus until I stumbled upon this link at the distillery.  Thanks again!

**Comment by jthomasmock** [comment]
+1 - thanks to Shannon for resharing- this is a great option!

**Reply by joelnitta** [reply]
Thanks for the comments! I’m very happy if it’s helpful for others. And super excited for the first comments on my (new) blog 😁

**Comment by drmowinckels** [comment]
I love this! Totally switching my blog and hugo template to accomodate it! Thanks @shannonpileggi for tweeting aboutit!

**Reply by drmowinckels** [reply]
Thanks to you, I am now also using giscus :D 
https://drmowinckels.io/blog/2021-03-23-using-github-actions-to-build-your-hugo-website/

**Reply by joelnitta** [reply]
Looks great, glad it's working for you!

**Comment by MaraAlexeev** [comment]
Thank you so much for writing this!! It will be a gem for the R community.

**Reply by joelnitta** [reply]
Thanks for the kind words :)
I really didn't do much, just put together some things others had already made. Major props to @laymonage for creating giscus in the first place and @MilesMcBain for figuring out how to get it (utterances, originally) to work with Distill.

**Comment by nxskok** [comment]
yes, it works! (After a little fiddling.) 

In the script in step 5, I (eventually) realized that something had to go in those [ENTER CATEGORY HERE] placeholders, and the specific things that had go in there were the things under Enable Giscus on the Giscus app page. (Your words "populated with your information" must have eluded me the first time.)

Thank you for inspiring me to try this, and, not only that, thanks to you, I was successful!

**Reply by joelnitta** [reply]
Yes it took me a little while playing with the Giscus app page to realize that. Thanks for the comment, hopefully that may clear it up for others as well.

**Reply by Nitinkumar-Gove** [reply]
From where did you get value for [your-category-id] ? I am unable to locate this during integration.

**Reply by joelnitta** [reply]
Do you mean this line in the HTML snippet above?

```
 script.setAttribute("data-category-id", "[ENTER CATEGORY ID HERE]");
```

That should show up in the giscus app once you've selected a category from the drop-down menu, like this: https://imgur.com/gallery/Md6ul0k

Did you select a category?

**Comment by L-Groeninger** [comment]
Wow, thanks a lot!:) I also managed to follow your tutorial and got it working

**Reply by joelnitta** [reply]
Yay! Glad to know it worked for you.

**Comment by loreabad6** [comment]
Thank you so much this is great!

**Reply by joelnitta** [reply]
You're welcome! 😄

**Comment by jeweljohnsonj** [comment]
Thank you for writing this article. It was very easy to follow through.

**Reply by joelnitta** [reply]
Thanks for the feedback, I'm glad it was useful!

**Comment by mfherman** [comment]
Thanks for writing this up -- so easy to set up on my Distill site!

One small thing that tripped me up is that if your blog posts are in a folder other than `posts/`, you have to modify the example `giscus.html` file to find posts in the non-standard directory. The small change for my site is [here](https://github.com/mfherman/mattherman-v2/blob/e526d333803f9faa11df757fe3acf4a726830197/giscus.html#L3). Thanks again!

**Reply by joelnitta** [reply]
Glad it worked for you!

Thanks for adding that clarification, I wouldn't have noticed otherwise.

**Reply by gustavdelius** [reply]
@mfherman, what modification would I have to make if my pages are in the root directory of the website. I tried to just take out the "/posts" part but that did not work.

**Comment by shamindras** [comment]
Thanks @joelnitta for these great instructions. I just switched to `giscus` from `utterances` on [my distill site](https://www.shamindras.com/). Was super easy thanks to you!

**Reply by joelnitta** [reply]
Awesome!

**Comment by be-favorite** [comment]
Thanks a lot!

**Reply by joelnitta** [reply]
You're welcome! 😊

**Comment by tongakuot** [comment]
Thank you so much, Joel! I have successfully added Giscus to my blog.

**Reply by joelnitta** [reply]
Awesome!

**Comment by gustavdelius** [comment]
Hi Joel. In your update to this blog post you mention Quarto. Did you switch from distill to Quarto and if so, what are the advantages?

**Reply by joelnitta** [reply]
I did switch to Quarto. I find the advantages broadly speaking that Quarto can do just about everything that was possible in Rmarkdown, but without lots of extra packages and work-arounds (such as the one described in this blogpost). I get the impression that the developers of quarto saw how Rmarkdown had evolved and decided to implement a lot of the functionality that had grown organically in Rmarkdown directly in Quarto. The other major feature is that it can run various languages natively (not just R), but I primarily use R so I don't notice that as much.

**Reply by gustavdelius** [reply]
I am very glad that your blog made me aware of Quarto. You are right that it is indeed the future of R Markdown.

**Comment by cpsyctc** [comment]
Thanks for this.  Enabled me to set it up on my blog (https://www.psyctc.org/Rblog/)  I suspect I'll remain the only person to comment there but at least the option is there!!  Very best, Chris
